stats/statsvariables.cpp

```diff
@@ -1681,6 +1681,58 @@ struct DayOfWeekVariable : public StatsVariableTemplate<StatsVariable::Type::Dis
 	}
 };
 
+// ============ Rating ============
+struct RatingBinner : public SimpleBinner<RatingBinner, IntBin> {
+	QString format(const StatsBin &bin) const override {
+		return QString("🌟").repeated(derived_bin(bin).value);
+	}
+
+	int to_bin_value(const dive *d) const {
+		int res = (int)d->rating;
+		return res;
+	}
+};
+
+static RatingBinner rating_binner;
+struct RatingVariable : public StatsVariableTemplate<StatsVariable::Type::Discrete> {
+	QString name() const override {
+		return StatsTranslations::tr("Rating");
+	}
+	QString diveCategories(const dive *d) const override {
+		int rating = (int)d->rating;
+		return QString("🌟").repeated(rating);
+	}
+	std::vector<const StatsBinner *> binners() const override {
+		return { &rating_binner };
+	}
+};
+
+// ============ Rating ============
+struct VisibilityBinner : public SimpleBinner<RatingBinner, IntBin> {
```

Comment:
I'm surprised that this compiles. This should read VisibilityBinner. I will have to add some TMP (template meta programming) to catch this kind of error.